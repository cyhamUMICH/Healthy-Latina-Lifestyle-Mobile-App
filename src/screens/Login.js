import React, { useState } from 'react';
import { View } from 'react-native';
import { styles } from '../styles/Styles';
import { colors } from '../styles/Colors';
import { Button } from 'react-native-elements';
import { TextInput } from 'react-native-gesture-handler';
import { Image } from 'react-native';
//import auth from '@react-native-firebase/auth';



//https://code.tutsplus.com/tutorials/common-react-native-app-layouts-login-page--cms-27639
const Login = (props) => {

const[username, setUsername] = useState("");
const[password, setPassword] = useState("");
const[needPassword, setNeed] = useState(false);

function handleLogon(){
//if username and password arte in the database,
//auth()
//.signInWithEmailAndPassword(username, password);
//else if they aren't add winvalid username/password.
setNeed(true);

}

function handlePasswordReset(){
props.navigation.navigate("Temp")
setNeed(false)

}
  return (
    <View style={styles.app}>


    <View style={styles.fullWidthWindow}>
      <View style={styles.smallerWidthWindow}>
        <Image 
          source={require('../../assets/logo-title.png')} 
          style={styles.loginPhoto}
          />

        <View style={styles.inputView}>
          <TextInput
            style={styles.inputText}
            textAlign="center"
            placeholder="username"
            placeholderTextColor={colors.text}
            onChangeText={username => setUsername(text)}
          />
  
          <TextInput
            style={styles.inputText}  
            textAlign="center"
            placeholder="password"
            placeholderTextColor={colors.text}
            secureTextEntry={true}
            onChangeText={(password) => setPassword(text)}
          />
        </View>

          <Button 
          buttonStyle={styles.smallButton}
          titleStyle={styles.buttonText}
          title="Login"  onPress={() => handleLogon()}
          />
          <Button 
          buttonStyle={styles.smallButton}
          titleStyle={styles.buttonText}
          title="Create an account" onPress={() => props.navigation.navigate("Register")} 
          />

          <View>
          {needPassword &&
          <Button
          buttonStyle={styles.smallButton}
          titleStyle={styles.buttonText}
          title="Forgot Password" onPress={() => handlePasswordReset()}
          />
          }
          </View>


        </View>
      </View>
    </View>
  );
};
export default Login;


