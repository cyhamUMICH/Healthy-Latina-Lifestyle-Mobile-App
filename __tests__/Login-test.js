// Code from: https://github.com/byCedric/expo-guide-ci/blob/master/__tests__/App-test.js
//      and https://jestjs.io/docs/tutorial-react-native
import React from 'react';
import Login from '../src/screens/Login';
import renderer from 'react-test-renderer';

test('Renders Login With the Loading Screen', async () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Login Without the Loading Screen', async () => {
  const tree = renderer.create(<Login skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});