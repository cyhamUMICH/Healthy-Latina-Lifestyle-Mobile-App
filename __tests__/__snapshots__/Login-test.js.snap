// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Login With the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": "2%",
        "paddingTop": "2%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "padding": "10%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "height": null,
            "justifyContent": "center",
            "width": null,
          }
        }
      />
      <View
        style={
          Object {
            "marginVertical": "2%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="username"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="password"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00A89D",
                "borderColor": "#2089dc",
                "borderRadius": 200,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": "2%",
                "marginHorizontal": "20%",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00A89D",
                "borderColor": "#2089dc",
                "borderRadius": 200,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": "2%",
                "marginHorizontal": "20%",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Create an account
            </Text>
          </View>
        </View>
      </View>
      <View />
    </View>
  </View>
</View>
`;

exports[`Renders Login Without the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": "2%",
        "paddingTop": "2%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "padding": "10%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "height": null,
            "justifyContent": "center",
            "width": null,
          }
        }
      />
      <View
        style={
          Object {
            "marginVertical": "2%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="username"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          collapsable={false}
          onChangeText={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          placeholder="password"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#00A89D",
              "borderRadius": 30,
              "marginBottom": "2%",
              "paddingHorizontal": "5%",
              "paddingVertical": "2%",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00A89D",
                "borderColor": "#2089dc",
                "borderRadius": 200,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": "2%",
                "marginHorizontal": "20%",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Login
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00A89D",
                "borderColor": "#2089dc",
                "borderRadius": 200,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": "2%",
                "marginHorizontal": "20%",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Create an account
            </Text>
          </View>
        </View>
      </View>
      <View />
    </View>
  </View>
</View>
`;
